### PR TITLE
Feature/login and register functionality

### PR DESCRIPTION
**Pull request by XanderSchoeman**
### **Overview:**
- Added posts methods for registering user and logging in/authenticating a user.
- Added get methods to get a user. Or getting users and viewing them and their necessary details in own DBMS presented through leaf html for easy management.

### CODE REVIEW

File: Sources/App/routes.swift [hunk 1]
```diff
@@ -1,98 +1,40 @@
...
+        let favouriteCount: Int = try req.content.syncGet(at: "favouriteCount")
+        let genrePref: String = try req.content.syncGet(at: "genrePref")
+        let userObj = User(id: nil, username: username, password: password, profileImage: profileImage, fullName: fullName, email: email, gender: gender, age: age, favouriteCount: favouriteCount, genrePref: genrePref)
+        return userObj.save(on: req)
+    }
     
+        router.post("authenticate") { req -> Future<[User]> in
+        let username: String = try req.content.syncGet(at: "username")
+        let password: String = try req.content.syncGet(at: "password")
+        let something = User.query(on: req).filter(\User.username == username).filter(\User.password == password).all()
```
Comment (Tiewhan) [inline Sources/App/routes.swift]: Name should be fixed. Try using foundUser or user instead of something

File: Resources/Views/home.leaf [hunk 2]
```diff
@@ -1,21 +1,22 @@
...
+<p>profileImage: <input type="text" name="profileImage" /></p>
+<p>fullName: <input type="text" name="fullName" /></p>
+<p>email: <input type="text" name="email" /></p>
+<p>gender: <input type="text" name="gender" /></p>
+<p>age: <input type="text" name="age" /></p>
+<p>favouriteCount: <input type="text" name="favouriteCount" /></p>
+<p>genrePref: <input type="text" name="genrePref" /></p>
+<p><input type="submit" /></p>
+</form>
+<h2>Users In DB…</h2>
```
Comment (Tiewhan) [inline Resources/Views/home.leaf:16]: Why are you creating an HTML page? APIs wouldn't use this at all

File: Sources/App/routes.swift (hunk 1 above)
Comment (XanderSchoeman) [reply Sources/App/routes.swift]: I completely forgot about that var, thanks for reminding me. (was testing around at some point).

File: Resources/Views/home.leaf (hunk 2 above)
Comment (XanderSchoeman) [reply Resources/Views/home.leaf:16]: The html page is strictly for my own use, as it allowed me to test the database and see which users are in it. This is for future use if anyone ever wanted to use the api.

**Review by Tiewhan — APPROVED** [review]

**Pull request merged**